File: q02/01.rb
#!/usr/bin/env ruby

op = ["+", "-", "*", "/", ""]
for i in 1000..9999 do
  c = i.to_s
  unless c.include?("0")
    for j in op do
      for k in op do
        for l in op do
          val = c[3] + j + c[2] + k + c[1] + l + c[0]
          if val.length > 4
            if i == eval(val)
            puts "#{val} = #{i}"
            end
          end
        end
      end
    end
  end
end

File: q02/01.js
var op = ["+", "-", "*", "/", ""];
// var op = ["*", ""];
for (var i = 1000; i < 10000; i++) {
  var c = String(i);
  for (var j = 0; j < op.length; j++) {
    for (var k = 0; k < op.length; k++) {
      for (var l = 0; l < op.length; l++) {
        val = c.charAt(3) + op[j] + c.charAt(2) + op[k] + c.charAt(1) + op[l] + c.charAt(0);
        if (val.length > 4) {
          if (i == eval(val)) {
            console.log(val + " = " + i);
          }
        }
      }
    }
  }
}